File: Problems/boj_회의실배정.py
#https://www.acmicpc.net/problem/1931
#Greedy, Sort

n = int(input())
l = [tuple(map(int, input().split())) for _ in range(n)]
l.sort(key=lambda x: (x[1], x[0])) #빨리 끝나는 시간으로 정렬해야 뒤에 더 고려해볼 회의가 많아 진다.

s, e = l[0][0], l[0][1]
count = 1
for j in range(1, n):
    ns, ne = l[j]
    if e <= ns:
        count += 1
        e = ne
    
print(count)

File: Problems/cpp/boj_회의실배정.cpp
//https://www.acmicpc.net/problem/1931
//Greedy, Sort


#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

bool cmp(const pair<int, int>& a, const pair<int, int>& b){
    if (a.second == b.second)
        return a.first < b.first;
    else
        return a.second < b.second;
}

int main()
{
    ios::sync_with_stdio(0);
    cin.tie(0);

    int n, s, e, result=1;
    vector<pair<int, int>> vec;
    cin >> n;

    
    for (int i=0; i<n; i++){
        cin >> s >> e;
        vec.push_back(make_pair(s, e));
    }

    sort(vec.begin(), vec.end(), cmp);

    int start = vec[0].second;
    for (int i=1; i<n; i++){
        if (start <= vec[i].first){
            result++;
            start = vec[i].second;
        }
    }

    cout << result;
}